pkg/providers/openstack/openstack_auth.go

```diff
@@ -0,0 +1,97 @@
+package openstack
+
+import (
+	"fmt"
+	"github.com/emc-advanced-dev/unik/pkg/config"
+	"github.com/rackspace/gophercloud"
+	"github.com/rackspace/gophercloud/openstack"
+	"os"
+)
+
+type OpenStackHandle struct {
+	AuthClient *gophercloud.ProviderClient
+	Region     string
+}
+
+func ObtainCredentials(conf config.Openstack) (*gophercloud.AuthOptions, error) {
+	// Obtain credentials passed in config file
+	auth := gophercloud.AuthOptions{
+		IdentityEndpoint: conf.OSAuthUrl,
+		UserID:           conf.OSUserId,
+		Username:         conf.OSUserName,
+		Password:         conf.OSPassword,
+		TenantID:         conf.OSTenantId,
+		TenantName:       conf.OSTenantName,
+		DomainID:         conf.OSDomainId,
+		DomainName:       conf.OSDomainName,
+	}
+
+	// Override with environment variables
+	if v := os.Getenv("OS_AUTH_URL"); v != "" {
+		auth.IdentityEndpoint = v
+	}
+	if v := os.Getenv("OS_USER_ID"); v != "" {
+		auth.UserID = v
+	}
+	if v := os.Getenv("OS_USERNAME"); v != "" {
+		auth.Username = v
+	}
+	if v := os.Getenv("OS_PASSWORD"); v != "" {
+		auth.Password = v
+	}
+	if v := os.Getenv("OS_TENANT_ID"); v != "" {
+		auth.TenantID = v
+	}
+	if v := os.Getenv("OS_TENANT_NAME"); v != "" {
+		auth.TenantName = v
+	}
+	if v := os.Getenv("OS_DOMAIN_ID"); v != "" {
+		auth.DomainID = v
+	}
+	if v := os.Getenv("OS_DOMAIN_NAME"); v != "" {
+		auth.DomainName = v
+	}
+
+	// Validate
```

Comment:
It's almost a copy-paste from `gophercloud` library - [link](https://github.com/rackspace/gophercloud/blob/master/openstack/auth_env.go#L36). Which, of course, doesn't imply that it's the best way to do. Suggestions?

Perhaps we should add `testConnection()` and then throw error if we couldn't connect?